### PR TITLE
Refs #29540 - Remove 3.16 deprecated functionality and warnings

### PR DESCRIPTION
**Pull request by jeremylenz**
There are deprecation warnings about functionality to be removed in 3.16. We need to remove the following in 3.16:

_Update 5/7/20:_ per @jlsherrill Puppet and OStree deprecations have been pushed to 4.0, as well as the background download type.  I am reverting those commits rather than removing them, so whoever ends up with that task can cherry-pick the originals later and get a bit of a head start.

~- [x] Background download type
Deprecation warning at /home/jomitsch/tool_belt/repos/katello/3.15.0/katello/app/lib/actions/katello/repository/update.rb line 16 is marked for removal in version 3.16, which is the next release version.~

~- [x] Puppet and ostree
Deprecation warning at /home/jomitsch/tool_belt/repos/katello/3.15.0/katello/app/lib/actions/katello/repository/update.rb line 20 is marked for removal in version 3.16, which is the next release version.~

~- [x] Background download type
Deprecation warning at /home/jomitsch/tool_belt/repos/katello/3.15.0/katello/app/lib/actions/katello/repository/create.rb line 14 is marked for removal in version 3.16, which is the next release version.~

~- [x] Puppet and ostree
Deprecation warning at /home/jomitsch/tool_belt/repos/katello/3.15.0/katello/app/lib/actions/katello/repository/create.rb line 18 is marked for removal in version 3.16, which is the next release version.~

- [x] Events API
Deprecation warning at /home/jomitsch/tool_belt/repos/katello/3.15.0/katello/app/controllers/katello/api/v2/host_subscriptions_controller.rb line 58 is marked for removal in version 3.16, which is the next release version.

~- [x] Puppet and ostree (create repo)
Deprecation warning at /home/jomitsch/tool_belt/repos/katello/3.15.0/katello/app/controllers/katello/api/v2/repositories_controller.rb line 223 is marked for removal in version 3.16, which is the next release version.~

### CODE REVIEW

File: app/controllers/katello/api/v2/host_subscriptions_controller.rb [hunk 1]
```diff
@@ -56,13 +56,6 @@ def auto_attach
       respond_for_index(:collection => index_response, :template => "index")
     end
 
-    api :GET, "/hosts/:host_id/subscriptions/events", N_("List subscription events for the host"), :deprecated => true
```
Comment (jturel) [inline app/controllers/katello/api/v2/host_subscriptions_controller.rb:59]: We should remove the route for this too

**Comment by jturel** [comment]
Looks like there's another API to remove: https://github.com/Katello/katello/blob/93fab731a53f9813a1b8d06fc35bfcf049f7770a/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L7

cc @mdellweg for awareness

**Comment by jeremylenz** [comment]
@jturel I was thinking squash and merge for this one, since that's turned on again. :wink:

**Comment by jturel** [comment]
> @jturel I was thinking squash and merge for this one, since that's turned on again. 😉

ACK. But that would contradict your comment (right?)

> I am reverting those commits rather than removing them, so whoever ends up with that task can cherry-pick the originals later and get a bit of a head start.

**Comment by jeremylenz** [comment]
@jturel The commits would still be available on the closed pull request, no?

**Comment by jturel** [comment]
> @jturel The commits would still be available on the closed pull request, no?

That never occurred to me! mind_blown.gif

**Comment by mdellweg** [comment]
> Looks like there's another API to remove:
> 
> https://github.com/Katello/katello/blob/93fab731a53f9813a1b8d06fc35bfcf049f7770a/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L7
> 
> https://github.com/Katello/katello/blob/93fab731a53f9813a1b8d06fc35bfcf049f7770a/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L69-L71
> 
> cc @mdellweg for awareness

The corresponding candlepin change is still waiting:
https://github.com/candlepin/subscription-manager/pull/2132

**Comment by jturel** [comment]
> > Looks like there's another API to remove:
> > https://github.com/Katello/katello/blob/93fab731a53f9813a1b8d06fc35bfcf049f7770a/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L7
> > 
> > https://github.com/Katello/katello/blob/93fab731a53f9813a1b8d06fc35bfcf049f7770a/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L69-L71
> > 
> > cc @mdellweg for awareness
> 
> The corresponding candlepin change is still waiting:
> [candlepin/subscription-manager#2132](https://github.com/candlepin/subscription-manager/pull/2132)

Thanks, guess we'll keep it for now. I guess we should update the deprecation warning for Katello 4.0 and try to get the sub-man PR merged in

File: app/controllers/katello/api/v2/host_subscriptions_controller.rb (hunk 1 above)
Comment (jeremylenz) [reply app/controllers/katello/api/v2/host_subscriptions_controller.rb:59]: took me a minute to find, but that's done now :)

**Comment by jturel** [comment]
I just had a look through the code and it appears we're caught up on deprecations with this PR which is awesome. I found this though:

```
app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb:      ::Foreman::Deprecation.api_deprecation_warning("The 'applicable_hosts' field will be removed from this API in Katello 2.0")
app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb:      ::Foreman::Deprecation.api_deprecation_warning("The 'applicable_hosts' field will be removed from this API in Katello 2.0")
```

I think these warnings can be removed - I think the associated parameter has been gone for a while

**Pull request merged**